core/src/main/java/org/mobilitydata/gtfsvalidator/notice/NoticeContainer.java

```diff
@@ -152,4 +174,195 @@ public String exportSystemErrors(boolean isPretty) {
+        ClassPath.from(ClassLoader.getSystemClassLoader()).getTopLevelClasses(noticePackageName)) {
+      clazz = noticeClass.load();
+      if (!ValidationNotice.class.isAssignableFrom(clazz)) {
+        continue;
+      }
+      if (clazz.isEnum()) {
+        continue;
+      }
+      if (clazz.getSimpleName().equals(NOTICE_CONTAINER_CLASS_SIMPLE_NAME)) {
+        continue;
+      }
+      if (clazz.getSimpleName().equals(VALIDATION_NOTICE_CLASS_SIMPLE_NAME)) {
+        continue;
+      }
+      toReturn.add(
+          noticeClass.getSimpleName(),
+          extractNoticeProperties(clazz)
+      );
+    }
+    return toReturn;
+  }
+
+  /**
+   * Extract information from notices defined in the main module.
+   *
+   * @param validatorPackageName the name of the package that contains the {@code GtfsValidator}s
+   * @return the {@code com.google.gson.JsonObject} that contains the {@code ValidationNotice} class
+   * simple name, and the type of each parameter
+   * @throws IOException if the attempt to read class path resources (jar files or directories)
+   * failed.
+   */
+  private static JsonObject extractMainNoticesProperties(String validatorPackageName)
+      throws IOException {
+    JsonObject toReturn = new JsonObject();
+    for (ClassPath.ClassInfo validatorClass : ClassPath.from(ClassLoader.getSystemClassLoader())
+            .getTopLevelClasses(validatorPackageName)) {
+      for (Class<?> innerNoticeClass : validatorClass.load().getDeclaredClasses()) {
+        if (ValidationNotice.class.isAssignableFrom(innerNoticeClass))
+        toReturn.add(
+            innerNoticeClass.getSimpleName(),
+            extractNoticeProperties(innerNoticeClass)
+        );
+      }
+    }
+    return toReturn;
+  }
+
+  /**
+   * Return a {@code JsonArray} that contains information about the type of each parameter of
+   * a {@code ValidationNotice} using the constructor of the class that is annotated by
+   * {@code  NoticeExport}.
+   *
+   * @param validationNoticeSubClass the {@code ValidationNotice} sub class to extract information
+   *                                from
+   * @return a {@code JsonArray} that contains information about the type of each parameter of
+   * said {@code ValidationNotice} using the constructor of the class that is annotated by
+   * {@code NoticeExport}.
+   */
+  private static JsonArray extractNoticeProperties(Class<?> validationNoticeSubClass) {
+    Constructor<?> constructor = getAnnotatedConstructor(validationNoticeSubClass);
+    JsonArray parametersAsJsonArray = new JsonArray();
+    Arrays.stream(constructor.getParameters()).forEach(parameter -> {
+      JsonObject parameterDetails = new JsonObject();
+      parameterDetails.addProperty(NAME, parameter.getName());
+        parameterDetails.addProperty(TYPE, mapDataType(parameter));
+      parametersAsJsonArray.add(parameterDetails);
+    });
+    if (!noticeConstructorHasSeverityLevelParameter(constructor)) {
```

Comment:
I'm not quite sure what is going on here. From my understanding, if a constructor has the severity level parameter, then you don't enter here, but in `NoticeContainerTest.java`, it looks like `SomeTestValidationNotice.java` has the severity level parameter in its constructor, but the property `{"SeverityLevel": "BLOB"}` is added to the notices schema (so it would have passed the condition at line 296). Am I missing something?

Reply:
By default, each `Notice` has a `SeverityLevel` whether it is explicitly passed in the constructor or not. Since we base the schema elaboration on the `Notice` constructor, we have to take this into account and it is the rationale behind this `if` statement.

Reply:
Trying to be more clear - let's say that we elaborate the schema based on `SomeTestValidationNotice.java` and `EmptyFileNotice.java`: 

The resulting schema should specify that the two notices contain a `SeverityLevel` object - however, as you mentioned `SomeTestValidationNotice.java` already has `SeverityLevel` in its constructor parameter -> nothing to do here. For `EmptyFileNotice.java` we have to add it manually; hence the extra step provided in the`if` statement.

Reply:
All good with this, but then why is the property `{"SeverityLevel": "BLOB"}` added to the notices schema for `SomeTestValidationNotice.java`?  From my understanding, it should not enter the `if` statement.

Reply:
Since `SeverityLevel` is one of the parameters of `SomeTestValidationNotice`'s constructor, it is added to the schema when evaluating lines 290-295
 https://github.com/MobilityData/gtfs-validator/blob/0b44560551b09033498d22ee35054b31a2edb114/core/src/main/java/org/mobilitydata/gtfsvalidator/notice/NoticeContainer.java#L290. 

> From my understanding, it should not enter the if statement.
Absolutely, in this case the `if` statement is not entered.